### PR TITLE
Feat/translation files

### PR DESCRIPTION
**Pull request by Rob117**
There were SO MANY CONFLICTS with main that I had to do this:
https://stackoverflow.com/questions/20229222/how-to-resolve-all-conflicts-using-head-with-any-mergetool

I thought I was clever by basing this branch off of the translation wrapper brranch. I was a FOOL because it caused divergent timelines on main, and I needed to resolve an unholy amount of conflicts.

Resolving these conflics took almost as long as writing the code in the first place.

### CODE REVIEW

File: utils/translations.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  },
+  featureSection: {
+    featureOne: '毎月、国産オーガニック・フリーレーンジマッシュルームがお家に届く',
+    featureTwo: '厳選されたマッシュルームの品質確認には、世界レベルのテスターを使うもの',
+    featureThree: '科学で実証された：毒は一切入っていない'
+  },
+  footer: { credits: '里咲、そして露舞' }
+};
+
+type MasterTranslationBlock = Record<SupportedLocale, LanguageSpecificTranslation>;
```
Comment (risafj) [inline utils/translations.ts:65]: You are such a TS nerd now

Comment (risafj) [reply utils/translations.ts:65]: Why is it "block"?

File: components/templates/login/LoginTemplate.tsx [hunk 2]
```diff
@@ -27,7 +27,7 @@ export const LoginTemplate: TopLevelTemplate = ({ locale }) => {
         </p>
 
         <LocalizedLink path='/'>
-          { "Or don't :(" }
+          { 'Or don\'t :(' }
```
Comment (risafj) [inline components/templates/login/LoginTemplate.tsx:30]: Are you not translating this one?

File: components/atoms/TranslatedText.tsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { Typography, TypographyProps  } from '@mui/material';
+import React from 'react';
+
+// TS error for components
+// https://github.com/mui-org/material-ui/issues/19512#issue-558587710
+
+type Props =  TypographyProps & { children: string, component?: any };
+export const TranslatedText = ({ children, ...rest } : Props) => {
```
Comment (risafj) [inline components/atoms/TranslatedText.tsx:8]: I think it's customary to do `...props` here?

```suggestion
export const TranslatedText = ({ children, ...props } : Props) => {
```

File: components/atoms/TranslatedText.tsx [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { Typography, TypographyProps  } from '@mui/material';
+import React from 'react';
+
+// TS error for components
+// https://github.com/mui-org/material-ui/issues/19512#issue-558587710
+
+type Props =  TypographyProps & { children: string, component?: any };
```
Comment (risafj) [inline components/atoms/TranslatedText.tsx:7]: Why `component`?
If it's a component, maybe it should be JSXElement or React.ReactNode.
If it's the `component` prop for the Typography, should it be a string?

File: components/page_wrappers/TranslationsWrapper.tsx [hunk 5]
```diff
@@ -1,15 +1,19 @@
...
 
-export const TranslationContext = React.createContext<{ locale: SupportedLocale }>({ locale: 'en' });
+export const TranslationContext = React.createContext<{
+  locale: SupportedLocale, translation: LanguageSpecificTranslation
+}>({ locale: 'en', translation: translations.en });
+
 export const TranslationsWrapper = ({ locale, children }: Props) => {  
   return (
-    <TranslationContext.Provider value={ { locale } } >
+    <TranslationContext.Provider value={ { locale, translation: translations[locale] } } >
```
Comment (risafj) [inline components/page_wrappers/TranslationsWrapper.tsx:16]: This whole thing is smart. Bravo!

**Review by risafj — CHANGES_REQUESTED** [review]
Great job Big Jam

File: components/atoms/TranslatedText.tsx (hunk 4 above)
Comment (Rob117) [reply components/atoms/TranslatedText.tsx:7]: Original component type is `any`, mirrored here

File: components/page_wrappers/TranslationsWrapper.tsx (hunk 5 above)
Comment (Rob117) [reply components/page_wrappers/TranslationsWrapper.tsx:16]: Thank you~

**Review by risafj — APPROVED** [review]

**Pull request merged**